tunnel/intercept/tproxy/tproxy_linux.go

```diff
@@ -122,10 +128,20 @@ func (self *interceptor) StopIntercepting(serviceName string, tracker intercept.
 		proxy := val.(*tProxy)
 		proxy.Stop(tracker)
 		self.serviceProxies.Remove(serviceName)
+		self.cleanupChains()
 	}
 	return nil
 }
 
+func (self *interceptor) cleanupChains() {
+	if self.serviceProxies.IsEmpty() {
+		deleteIptablesChain(self.ipt, mangleTable, "PREROUTING", dstChain)
```

Comment:
is there any sort of reference counting done (it's not clear in just this review) that some other service doesn't use the same hostname? it's perfectly fine for two services from two identities to use "my-intercept" as long as both identities were to use different ports. same is true for one identity with multiple services using "my-intercept". you probably do not want to delete the rules unless ALL the services using that host are gone

Reply:
This is cleaning up the chains that are used by all services, so we only remove them if we have zero services being intercepted. 
The other rules are per-port range, so we remove those as specific port ranges are removed. 
Allocated IP addresses are per DNS entry and those are reference counted using the address tracker.